cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -344,4 +349,17 @@ public void removeStreamTags(HttpRequest request, HttpResponder responder,
   public void recordRun(HttpRequest request, HttpResponder responder) {
     responder.sendString(HttpResponseStatus.OK, "Metadata recorded successfully");
   }
+
+  // *** Search endpoints ***
+
+  @GET
+  @Path("/namespaces/{namespace-id}/metadata/search")
+  public void searchMetadata(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @QueryParam("query") String searchQuery,
+                             @QueryParam("target") MetadataSearchTargetType target) throws Exception {
+    Iterable<MetadataSearchResultRecord> results = metadataAdmin.searchMetadata(searchQuery, target);
+
+    responder.sendJson(HttpResponseStatus.OK, Sets.newHashSet(results));
```

Comment:
Not a big deal, but we had used ``Set`` because the results from all these APIs are collections with unique elements.